lib/storage/index.ts

```diff
@@ -1,3 +1,190 @@
-import WebStorage from './WebStorage';
+import * as Logger from '../Logger';
 
-export default WebStorage;
+import PlatformStorage from './platforms';
+import InstanceSync from './InstanceSync';
+import NoopProvider from './providers/NoopProvider';
+import type StorageProvider from './providers/types';
+
+let provider = PlatformStorage;
+let shouldKeepInstancesSync = false;
+let finishInitalization: (value?: unknown) => void;
+const initPromise = new Promise((resolve) => {
+    finishInitalization = resolve;
+});
+
+type Storage = {
+    getStorageProvider: () => StorageProvider;
+} & Omit<StorageProvider, 'name'>;
+
+/**
+ * Degrade performance by removing the storage provider and only using cache
+ */
+function degradePerformance(error: Error) {
+    Logger.logAlert(`Error while using ${provider.name}. Falling back to only using cache and dropping storage.`);
+    console.error(error);
+    provider = NoopProvider;
```

Comment:
@tgolen @marcaaron also a question about this - do we want to fallback to `NoopProvider` (does nothing) or to `MemoryProvider` stores all data in RAM?

Reply:
I think the memory provider would be better. That's what the log message assumes.